src/main/java/pl/edu/zut/mad/schedule/search/ScheduleSpecification.java

```diff
@@ -0,0 +1,32 @@
+package pl.edu.zut.mad.schedule.search;
+
+import org.springframework.data.jpa.domain.Specification;
+import pl.edu.zut.mad.schedule.model.inner.Schedule;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+
+import static pl.edu.zut.mad.schedule.model.inner.Schedule.Fields.DATE;
+
+public class ScheduleSpecification implements Specification<Schedule> {
+
+    private SearchCriteria searchCriteria;
+
+    ScheduleSpecification(SearchCriteria searchCriteria) {
+        this.searchCriteria = searchCriteria;
+    }
+
+    @Override
+    public Predicate toPredicate(Root<Schedule> root, CriteriaQuery<?> query, CriteriaBuilder cb) {
+
```

Comment:
When I seen variable `cb` I have to check method parameters and check what type is it so full name or even `criteria` would be better.